src/main/webapp/app/entities/text-exercise/text-exercise-update.component.ts

```diff
@@ -1,51 +1,136 @@
 import { Component, OnInit } from '@angular/core';
-import { ActivatedRoute } from '@angular/router';
+import { ActivatedRoute, Router } from '@angular/router';
 import { HttpErrorResponse, HttpResponse } from '@angular/common/http';
-import { Observable } from 'rxjs';
 
+import { Observable } from 'rxjs/Observable';
+import { JhiAlertService, JhiEventManager } from 'ng-jhipster';
+
+import { TextExercise } from './text-exercise.model';
 import { TextExerciseService } from './text-exercise.service';
-import { TextExercise } from 'app/entities/text-exercise/text-exercise.model';
+import { Course, CourseService } from '../course';
+
+import { ExerciseCategory, ExerciseService } from 'app/entities/exercise';
+import { ExampleSubmissionService } from 'app/entities/example-submission/example-submission.service';
+import { KatexCommand } from 'app/markdown-editor/commands';
+import { EditorMode } from 'app/markdown-editor';
+import { MAX_SCORE_PATTERN } from 'app/app.constants';
+import { AssessmentType } from 'app/entities/assessment-type';
 
 @Component({
     selector: 'jhi-text-exercise-update',
     templateUrl: './text-exercise-update.component.html',
+    styleUrls: ['./text-exercise-update.scss'],
 })
 export class TextExerciseUpdateComponent implements OnInit {
+    EditorMode = EditorMode;
+    AssessmentType = AssessmentType;
+
     textExercise: TextExercise;
     isSaving: boolean;
+    maxScorePattern = MAX_SCORE_PATTERN;
+    exerciseCategories: ExerciseCategory[];
+    existingCategories: ExerciseCategory[];
+    notificationText: string | null;
+
+    courses: Course[];
+
+    domainCommandsProblemStatement = [new KatexCommand()];
+    domainCommandsSampleSolution = [new KatexCommand()];
+    domainCommandsGradingInstructions = [new KatexCommand()];
 
-    constructor(private textExerciseService: TextExerciseService, private activatedRoute: ActivatedRoute) {}
+    constructor(
+        private jhiAlertService: JhiAlertService,
+        private textExerciseService: TextExerciseService,
+        private exerciseService: ExerciseService,
+        private courseService: CourseService,
+        private eventManager: JhiEventManager,
+        private exampleSubmissionService: ExampleSubmissionService,
+        private activatedRoute: ActivatedRoute,
+        private router: Router,
+    ) {}
 
     ngOnInit() {
-        this.isSaving = false;
+        // This is used to scroll page to the top of the page, because the routing keeps the position for the
+        // new page from previous page.
+        window.scroll(0, 0);
```

Comment:
I would always prefer to use our `window.service.ts`, so that we can inject the window reference if we need to (e.g. in our tests).

Reply:
Good suggestion. Done